assets/js/admin/meta-boxes-subscription.js

```diff
@@ -340,7 +340,7 @@ jQuery( function ( $ ) {
 		}
 	} );
 
-	$( 'body.post-type-shop_subscription #post' ).on( 'submit', function () {
+	$( 'body.post-type-shop_subscription #post,body.woocommerce_page_wc-orders--shop_subscription #order' ).on( 'submit', function () {
```

Comment:
```suggestion
	$( 'body.post-type-shop_subscription #post, body.woocommerce_page_wc-orders--shop_subscription #order' ).on( 'submit', function () {
```
Nit: keep space between selectors for consistency with selectors above.